src/Microsoft.Framework.Runtime.Common/Impl/RuntimeEnvironment.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.Versioning;
+
+namespace Microsoft.Framework.Runtime
+{
+    public class RuntimeEnvironment : IRuntimeEnvironment
+    {
+#if DNXCORE50
+        private static readonly Version _osVersion = WindowsApis.OSVersion;
+#else
+        private static readonly Version _osVersion = Environment.OSVersion.Version;
+#endif
+
+        private static readonly OperatingSystem _operatingSystem;
+
+        static RuntimeEnvironment()
+        {
+#if DNXCORE50
+            // TODO: Weak assumption, will not work well on non Windows
+            // waiting for the CLR team to provide the right API
+            _operatingSystem = OperatingSystem.Windows;
+#else
+            Version osVersion = Environment.OSVersion.Version;
+            switch (Environment.OSVersion.Platform)
+            {
+                case PlatformID.Win32NT:
+                    _operatingSystem = OperatingSystem.Windows;
+                    break;
+                case PlatformID.Unix:
+                    _operatingSystem = OperatingSystem.Unix;
+                    break;
+                case PlatformID.MacOSX:
```

Comment:
This doesn't work on mono: https://github.com/mono/mono/blob/master/mcs/class/corlib/System/Environment.cs#L238